omod-dstu3/src/main/java/org/openmrs/module/fhir2/providers/r3/EncounterFhirResourceProvider.java

```diff
@@ -53,6 +53,16 @@ public Encounter getEncounterById(@IdParam @NotNull IdType id) {
 		return VersionConvertor_30_40.convertEncounter(encounter);
 	}
 	
+	@History
+	@SuppressWarnings("unused")
+	public List<Resource> getEncounterHistoryById(@IdParam @NotNull IdType id) {
+		org.hl7.fhir.r4.model.Encounter encounter = encounterService.getEncounterByUuid(id.getIdPart());
+		if (encounter == null) {
+			throw new ResourceNotFoundException("Could not find encounter with Id " + id.getIdPart());
+		}
+		return VersionConvertor_30_40.convertEncounter(encounter).getContained();
+	}
```

Comment:
@ibacher Test for all these functions fail. Can you give an idea, what's wrong here

Reply:
Likely because the converter doesn't handled the contained resources properly, so when `getContained()` gets called here, it returns an empty list. We may need to look at implementing re-implementing the code to get the provenance resources as part of the R3 submodule.